doc/api/buffer.md

```diff
@@ -402,7 +402,7 @@ changes:
 
 > Stability: 0 - Deprecated: Use [`Buffer.from(buffer)`] instead.
 
-* `buffer` {Buffer} An existing `Buffer` to copy data from.
+* `buffer` {Buffer|Uint8Array} An existing `Buffer` to copy data from.
```

Comment:
This should likely be `Buffer|TypedArray` with the description `An existing `Buffer` or `TypedArray` instance to copy from.

It's worth noting that any `TypedArray` can be used, not just `Uint8Array`. The issue, however, is that as values are copied, each will be truncated to an 8-byte value. For instance, `Buffer.from(new Uint32Array([0xffff, 2, 3])` will produce a `Buffer` with values `[0xff, 2, 3]`.

Reply:
@jasnell That’s more or less already documented as https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_array – we accept anything with a `length` property, basically.

Reply:
Oh sorry, I meant just worth noting in this thread, not in the docs. My mistake, I wasn't clear about that at all. The change here that I'd like to see is just listing `TypedArray` instead of `Uint8Array` in the bullet point.

Reply:
@jasnell the body language may need some clarification if the signature mentions general Typed Arrays:

> Copies the passed `buffer` data onto a new `Buffer` instance.

For a typed array that isn't a `Uint8Array`, does the "data" refer to the typed values or the `.buffer` data?  e.g.

```js
Buffer.from(new Float32Array([Math.PI]))        // <Buffer 03>
Buffer.from(new Float32Array([Math.PI]).buffer) // <Buffer db 0f 49 40>
```

This issue is obviated in the Uint8Array case but is relevant for other typed arrays.

Reply:
Yeah, exactly: Other kinds of `TypedArray`s are treated just like regular `Array`s. Hence my comment above – I think it would make more sense to document `TypedArray` as part of the `Buffer.from(array)` heading.

Reply:
The `Buffer.from(new Float32Array([Math.PI]).buffer)` variant is already covered in the documentation by the `Class Method: Buffer.from(arrayBuffer[, byteOffset[, length]])` section. In that specific case, when an `ArrayBuffer` instance is passed, the `Buffer` returned is a view over the existing `ArrayBuffer` and no data is copied or coerced.

The `Buffer.from(new Float32Array([Math.PI]))` case is covered by the `Class Method: Buffer.from(buffer)` variant but the doc for that specific variant only needs to be indicate that any `TypedArray` can be passed but that each typed value will be copied and coerced into 8-bit values in the returned `Buffer`.

The `Class Method: Buffer.from(buffer)` and `Class Method: Buffer.from(array)` variants *can* be merged into a single section as the behavior for each is generally identical.

Reply:
Local state was messed up, resubmitted PR in https://github.com/nodejs/node/pull/19949 .  I agree with @addaleax 's suggestion to clarify the typed array situation elsewhere, but that is a separate discussion

Reply:
Also note that `Buffer.from(new Uint8Array([1,2,3]))` is treated exactly the same as `Buffer.from([1,2,3])` ... that is, the new `Buffer` contains a *copy* of the original.

while `Buffer.from((new Uint8Array([1,2,3])).buffer)` returns a new `Buffer` that contains a view over the same backing `ArrayBuffer`. No data is copied.